pygac/pod_reader.py

```diff
@@ -347,8 +348,10 @@ def _validate_tbm_header(cls, potential_tbm_header):
             return potential_tbm_header.copy()
 
         # This will raise a DecodingError if the data_set_name is not valid.
-        cls._decode_data_set_name(data_set_name)
-        return potential_tbm_header.copy()
+        data_set_name = cls._decode_data_set_name(data_set_name)
+        tbm_header = potential_tbm_header.copy()
+        tbm_header["data_set_name"] = data_set_name.decode().strip("\x80").encode()
```

Comment:
Could the `.decode().strip().encode()` part be moved to `_decode_data_set_name`?